src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/RssCheckingActor.java

```diff
@@ -32,17 +31,13 @@ public static Props props(ActorRef postStoringActor, SyndFeedProducer syndFeedFa
     @SuppressWarnings("unchecked")
     private void executeAction(ActorRef postStoringActor, SyndFeedProducer syndFeedFactory,
                                RssLink rssLink) {
-        try {
-            SyndFeed feed = syndFeedFactory.createFor(rssLink.getUrl());
-            List<SyndEntry> posts = feed.getEntries();
-            posts.forEach(post -> {
+        syndFeedFactory.createFor(rssLink.getUrl()).ifPresent(feed -> {
```

Comment:
Do we really need _lambda_ expression here? _Lambdas_ should make the code better shorter and better to read. In this case it's opposite. IMHO simple _if_ expression will make the code better to read, additionally _lambdas_ are terrible to debug. Maybe something like this:
```
Optional<SyndFeed> feed = syndFeedFactory.createFor(rssLink.getUrl());
if (feed.isPresent) {
    List<SyndEntry> posts = feed.getEntries();
    posts.forEach(post -> {
             RssEntryWithAuthor msg = new RssEntryWithAuthor(rssLink.getOwner(), post);
             postStoringActor.tell(msg, self());
         }
    );
}
```

WDYT?